Add/add developers tab to simple sites

roo2 opened the pull request:
Promo tab to show on simple sites.

currently a WIP

Target:
<img width="1622" alt="326636637-52a94270-7552-47b8-a232-95cf8abae1b4" src="https://github.com/Automattic/wp-calypso/assets/22446385/38423d9c-be43-4549-b29b-a66a70b488ea">
Current:
<img width="1009" alt="Screenshot 2024-05-03 at 11 28 22 am" src="https://github.com/Automattic/wp-calypso/assets/22446385/8f5fb61a-4461-4245-a8bc-acacc6600692">

roo2 committed:
add developer tab as promo

roo2 committed:
fix

davemart-in commented:
Nicely done @roo2! A couple of things:

1. Mind adding this developer icon to the dev tools tab?

![CleanShot 2024-05-02 at 22 03 51@2x](https://github.com/Automattic/wp-calypso/assets/5634774/d6a1764e-bcfa-4490-b1c1-f00d3cda90de)

Here's the SVG:

![developer](https://github.com/Automattic/wp-calypso/assets/5634774/79ddb2ec-a764-4dee-aea9-c7c7165465ae)

2. Here are the links:

- Upgrade now: https://wordpress.com/checkout/{SITE}/business
- Browse plugins: https://wordpress.com/plugins/{SITE}
- Hosting configuration: https://wordpress.com/support/hosting-configuration/
- Monitoring: https://developer.wordpress.com/docs/troubleshooting/site-monitoring/#metrics
- PHP logs: https://developer.wordpress.com/docs/troubleshooting/site-monitoring/#php-logs-and-webserver-logs
- Server Logs: https://developer.wordpress.com/docs/troubleshooting/site-monitoring/#php-logs-and-webserver-logs
- GitHub Deployments: https://developer.wordpress.com/docs/developer-tools/github-deployments/

lucasmendes-design commented:
It's looking great. Just some details: 

- It's missing the badge
- The title is too bold ((Hosting configuration, Monitoring, PHP logs, Server Logs, GitHub Deployments).

<img width="1248" alt="Screenshot 2024-05-06 at 14 12 46" src="https://github.com/Automattic/wp-calypso/assets/98343583/c5af67a2-1ff7-46d1-99f7-61e5ec07d23d">

roo2 committed:
add links, add dev tools icon

roo2 commented:
I've added the links, added the icon and updated the header weight to be 500.

<img width="1715" alt="Screenshot 2024-05-07 at 10 39 03 am" src="https://github.com/Automattic/wp-calypso/assets/22446385/8cf5d7da-5af3-4ab9-9703-1d2e340a77da">

lucasmendes-design commented:
Perfect :) 

Thanks @roo2!

allilevine commented:
I took a look at the copy re [this comment](https://github.com/Automattic/dotcom-forge/issues/6752#issuecomment-2096523627). Suggestions for accuracy:

> Optimize your site's performance and security by tailoring your server settings to your specific needs.

Access your site's database and tailor your server settings to your specific needs.

> Proactively monitor site health, detect issues early, and maintain a smooth user experience with instant alerts.

Proactively monitor your site's performance, including requests per minute and average response time.

> Quickly diagnose and resolve PHP issues with detailed error insights, enhancing site reliability.

View and download PHP error logs to quickly diagnose and resolve issues.

> Gain full visibility into server activity, helping you manage traffic and spot security issues early.

✅ 

> Automate updates from GitHub to streamline workflows, reduce errors, and enable faster deployments.

✅

provenself commented:
Editorial checking in via [Alli's label](https://github.com/Automattic/wp-calypso/pull/90246#issuecomment-2099144960). Only suggestion is:

> View and download PHP error logs to quickly diagnose and resolve issues.

View and download PHP error logs to diagnose and resolve issues quickly.

roo2 commented:
awesome, I think I'll merge it tomorrow evening with any copy revisions, and then if there are more revisions we can revise it.

arthur791004 commented on client/sites-dashboard-v2/site-preview-pane/dotcom-preview-pane.tsx:
```diff
@@ -45,10 +46,20 @@ const DotcomPreviewPane = ( {
+				<span>
+					{ __( 'Dev Tools' ) } <DevToolsIcon />
+				</span>,
+				! isAtomicSite,
```
Do we want to display it on a self-hosted site?

roo2 replied:
Ahh good question, at first I thought we do want it but reading the issue again, it says 

> Jetpack: keep the same behavior as now.
6752-gh-Automattic/dotcom-forge

So I think for jetpack we just need to have the one single panel right @allilevine ?

arthur791004 commented on client/dev-tools-promo/index.tsx:
```diff
@@ -0,0 +1,19 @@
+	page(
+		'/dev-tools-promo/:site',
+		siteSelection,
+		navigation,
```
We have to redirect to the Overview tab when the selected site changes to the atomic site 🙂

roo2 replied:
ahh I understand, thanks

arthur791004 commented on client/dev-tools-promo/components/dev-tools-promo.tsx:
```diff
@@ -0,0 +1,96 @@
+	const siteId = useSelector( getSelectedSiteId );
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) ) || '';
+
+	const PromoCard = ( { title, text, href }: PromoCardProps ) => (
```
[nit] It would be better to define another component outside the component 🙂

arthur791004 commented on client/dev-tools-promo/components/dev-tools-promo.tsx:
```diff
@@ -0,0 +1,96 @@
+			</div>
+			<div className="dev-tools-promo__cards">
+				<PromoCard
+					title={ translate( 'Hosting Configuration' ) }
```
[nit] The structure of these cards is the same, so it would be better to define an array and loop through it. But it's up to you 🙂

roo2 replied:
ahh fair.

arthur791004 commented on client/dev-tools-promo/components/dev-tools-promo.tsx:
```diff
@@ -0,0 +1,96 @@
+		<Card className="dev-tools-promo__card">
+			<CardHeading>{ title }</CardHeading>
+			<p>{ text }</p>
+			<a href={ href }>{ translate( 'Learn more' ) }</a>
```
How about opening the doc by the modal or Help Center? We can iterate it later 🙂

roo2 replied:
hmm you mean it would open in the help center modal and remain in calypso? What do you think @lucasmendes-design

arthur791004 replied:
YES! People may just want to know more about these features but it seems not friendly if they have to go back and forth. It's not a blocker

davemart-in replied:
+1 to opening docs links in the help center modal.

roo2 committed:
address feedback

roo2 commented:
thanks for the feedback @arthur791004  and @davemart-in I've fixed the issues and ready for another look:

* updated copy
* redirect to /hosting/ if the site isn't simple
* use an array and map to define promo cards
* open the links in help centre.

Only issue I see is that the help centre doesn't support anchor links very well so the scroll position is off on some pages.

roo2 committed:
fix refresh

roo2 commented on client/dev-tools-promo/index.tsx:
```diff
@@ -0,0 +1,30 @@
+	const state = context.store.getState();
+	const site = getSelectedSite( state );
+	if ( site && site.jetpack ) {
+		return page.redirect( `/hosting/${ context.params.site }` );
```
I saw that the redirect was done differently in `redirectHomeIfIneligible`, they use `context.page.replace` to redirect. One cool side effect of `page.redirect` here is that it seems to "remember" that you were supposed to be on the dev tools tab.

e.g. if you're on the dev tools tab for SITE_A, then you change to an atomic site SITE_B, it would redirect you to the `/hosting/SITE_B` page. then if you go to another simple site SITE_C, it would take you back to the dev tools tab for that site `/dev-tools-promo/SITE_C`.

The context.page.replace approach would have taken you to `/hosting/SITE_C`

arthur791004 reviewed (approved):
Tested and it works well. Nice job 🎉

arthur791004 commented:
> Only issue I see is that the help centre doesn't support anchor links very well so the scroll position is off on some pages.

The reason seems to be sticky elements + content height (it's not fully rendered yet). But it's out of scope.

candy02058912 reviewed (approved):
I tested it, and one thing I noticed is that when I click on Upgrade now > Checkout > Use the back button on the checkout page, it redirects to `/plans/:site`.

Otherwise it's working great, and I think ☝️ is a bit out of scope for this PR.

cleacos commented:
👍 Good improvement to allow custom tabs to be labeled with icons and for your future dropdown submenu integrated inside the tabs.

It will be useful for A4A too.

keoshi commented:
Yes yes, this is great!
